app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java

```diff
@@ -24,7 +72,316 @@ public View onCreateView(@NonNull LayoutInflater inflater,
+
+            // Genre to entries map
+            List<String> genres = entry.getAnime().getGenres();
+            for (String genre: genres) {
+                if (!genreToList.containsKey(genre)) genreToList.put(genre, new ArrayList<>());
+                genreToList.get(genre).add(entry);
+            }
+        }
+
+        // Find min/max years
+        List<Integer> years = new ArrayList<>(yearToList.keySet());
+        Collections.sort(years);
+        minYear = years.get(0);
+        maxYear = years.get(years.size() - 1);
+
+        // Generate the charts
+        setOverview();
+        setChartActivity();
+        setChartGenre();
+        setChartGenrePref();
+        setGenreBreakdownByYear();
+        setTop5();
+    }
+
+    /* Overview [Score Cards] */
+    public void setOverview() {
+        // Total animes watched
+        binding.tvCount.setText(String.format(Locale.getDefault(), "%d", entries.size()));
+
+        // Average rating
+        double sumRating = 0;
+        for (Entry entry: entries) sumRating += entry.getRating();
+        double rating = sumRating / entries.size();
+        binding.tvRating.setText(String.format(Locale.getDefault(), "%.1f", rating));
+
+        // Animes watched this year + average rating this year
+        int currentYear = Calendar.getInstance().get(Calendar.YEAR);
+        int prevYear = currentYear - 1;
+
+        // Default values
+        binding.tvRecentCount.setText("0");
+        binding.tvRecentRating.setText("n/a");
+        binding.tvCountCompare.setText("\u2195 n/a");
+        binding.tvRatingCompare.setText("\u2195 n/a");
+        binding.tvCountCompare.setTextColor(white);
+        binding.tvRatingCompare.setTextColor(white);
+
+        // Set current year info
+        List<Entry> recentEntries = new ArrayList<>();
+        if (yearToList.containsKey(currentYear)) {
+            recentEntries = yearToList.get(currentYear);
+            binding.tvRecentCount.setText(String.format(Locale.US, "%d", recentEntries.size()));
+            binding.tvRecentRating.setText(String.format(Locale.US, "%.1f", getAverageRating(recentEntries)));
+        }
+
+        // Set comparisons to previous year
+        if (yearToList.containsKey(prevYear)) {
+            List<Entry> prevEntries = yearToList.get(prevYear);
+            setComparisonText(prevEntries.size(), recentEntries.size(), binding.tvCountCompare);
+            if (!binding.tvRecentRating.getText().equals("n/a")) {
+                setComparisonText(getAverageRating(prevEntries),
+                        Double.parseDouble(binding.tvRecentRating.getText().toString()),
+                        binding.tvRatingCompare);
+            }
+        }
+    }
+
+    /* Activity (Line Chart) - year vs. entry count */
+    public void setChartActivity() {
+        LineChart chart = binding.chartActivity;
+
+        // Create data points
+        List<com.github.mikephil.charting.data.Entry> chartEntries = new ArrayList<>();
```

Comment:
nit: can you import this directly above and use `List<Entry>` here instead?

Reply:
I tried doing that, but it coincides with my own class named `Entry`, and it seems like Java doesn't have an import aliasing mechanism.